python/fbprophet/tests/test_prophet.py

```diff
@@ -28,14 +28,59 @@
 
 class TestProphet(TestCase):
 
+    @staticmethod
+    def rmse(predictions, targets):
+        return np.sqrt(np.mean((predictions - targets) ** 2))
+
     def test_fit_predict(self):
+        days = 30
         N = DATA.shape[0]
-        train = DATA.head(N // 2)
-        future = DATA.tail(N // 2)
+        train = DATA.head(N - days)
+        test = DATA.tail(days)
+        test.reset_index(inplace=True)
 
         forecaster = Prophet()
-        forecaster.fit(train)
-        forecaster.predict(future)
+        forecaster.fit(train, seed=1237861298)
+        np.random.seed(876543987)
+        future = forecaster.make_future_dataframe(days, include_history=False)
+        future = forecaster.predict(future)
+        # this gives ~ 10.64
+        res = self.rmse(future['yhat'], test['y'])
+        self.assertTrue(15 > res > 5)
+
+    def test_fit_predict_newton(self):
+        days = 30
+        N = DATA.shape[0]
+        train = DATA.head(N - days)
+        test = DATA.tail(days)
+        test.reset_index(inplace=True)
+
+        forecaster = Prophet()
+        forecaster.fit(train, seed=1237861298, algorithm='Newton')
+        np.random.seed(876543987)
+        future = forecaster.make_future_dataframe(days, include_history=False)
+        future = forecaster.predict(future)
+        # this gives ~ 10.64
+        res = self.rmse(future['yhat'], test['y'])
+        self.assertAlmostEqual(res, 23.44, places=2)
+
+    def test_fit_sampling_predict(self):
```

Comment:
How slow is this test? Is there a way to denote that a test is slow and show only be run optionally?

Reply:
Yes, there is a way of marking the test as slow, yet I am not sure how to pass that around via  setup.py `test` command. I need to look into it.

Reply:
Sampling test will now run only when `--test-slow` argument present when running tests